lib/MemoryModel/MemModel.cpp

```diff
@@ -189,7 +189,7 @@ bool SymbolTableInfo::computeGepOffset(const llvm::User *V, LocationSet& ls) {
         // Handling array types, skipe array handling here
         // We treat whole array as one, then we can distinguish different field of an array of struct
         // e.g. s[1].f1 is differet from s[0].f2
-        if(isa<ArrayType>(*gi))
+        if(isa<ArrayType>(gi.getIndexedType()))
```

Comment:
Hi Jared,

Thanks for your efforts in writing a bridgeGepIterator.

I found an issue that the iterator seems not overloading "operator++"?

Could you please try the following case
```
struct MyStruct {
        int * f1; 
        int * f2; 
};

int main() {
        struct MyStruct s[2];
        int a,b;
        s[0].f1 = &a; 
        s[1].f1 = &b; 

        // Different fields of different elements in a 
        // certain array are treated as different objects.
        NOALIAS(s[a].f1, s[b].f2);

        return 0;
}
```

After you compile the above one into bc with "-mem2reg" turned on, please take a look at the following value, which corresponds "s[a]" in the source code  
```
%arrayidx3 = getelementptr inbounds [2 x %struct.MyStruct], [2 x %struct.MyStruct]* %s, i64 0, i64 %idxprom, !dbg !132
```
Ideally, the gepIterator should return two types after the two loop iterations:
[2 x %struct.MyStruct]* 
[2 x %struct.MyStruct]

However, your iterator returns the same type during two loop iterations
[2 x %struct.MyStruct]* 

Could you please double check it?

Thanks